canonical-web-and-design/maas-ui
# feat(ui): close custom source connection form when successfully connected

## Done

- Rearranged/renamed some of the files for the Custom source portion of the Images page
- Made it so that the Connection form closes once images have been successfully fetched

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to /MAAS/r/images and select "Custom" for the source
- Enter a valid url for images, e.g. https://images.maas.io/ephemeral-v3/stable/, and click "Connect"
- Check that the form closes, and the url you used is displayed.
- Check that images were successfully added to state, in `bootresource.fetchedImages`

## Fixes

Fixes canonical-web-and-design/app-squad#111
